packages/web-client/app/utils/web3-strategies/layer2-chain.ts

```diff
@@ -144,7 +143,16 @@ export default abstract class Layer2ChainWeb3Strategy
       }
       this.onDisconnect();
     });
-    await this.provider.enable();
+
+    yield this.provider.enable();
+
+    // we only get here if the user has an account
+    // we wait for the logic in accountsChanged to run and fetch the user's depots
+    // before considering initialization done
+    // the connector is from WalletConnect and knows of accounts before the
+    // the accountsChanged event is triggered
+    yield this.waitForAccountDeferred.promise;
+    this.isInitializing = false;
```

Comment:
Just realized we can actually move this into the `accountsChanged` block since we have a different way of setting `isInitializing` for the non-reconnecting case. Will move this in, so the code is more straightforward.